src/firebase/firebase.js

```diff
@@ -1,94 +1,73 @@
-      let errorMsg = "";   
-      return new Promise((resolve, reject) => {
-        this.auth.signInWithEmailAndPassword(email, password).catch(function(error) {
-          let errorCode = error.code;
-          let errorMessage = error.message;
-          if (errorCode === 'auth/wrong-password') {
-            alert('Wrong password.');
-            errorMsg = "Wrong password";
-          } else {
-            alert(errorMessage);
-            errorMsg = errorMessage;
-          }
-          reject(errorMsg);
-          console.log(error);
-        }).then(() => {
-          resolve(true)
-        });
-      });
+export const doSignInWithEmailAndPassword = (email, password) => {
+  let errorMsg = "";
+  return new Promise((resolve, reject) => {
+  firebase.auth().signInWithEmailAndPassword(email, password).catch(function(error) {
+    const {code, message} = error;
+    if (code === 'auth/wrong-password') {
+      alert('Wrong password.');
+      errorMsg = "Wrong password";
+    } else {
+      alert(message);
+      errorMsg = message;
     }
+    reject(errorMsg);
+  }).then(() => {
+    resolve(true);
+  });
+  });
+}
 
-    doSignOut = () => {
-        return this.auth.signOut();
-    }
+export const doSignOut = () => {
+  return firebase.auth().signOut();
+}
 
-    isSignedIn = () => {
-        return this.auth.currentUser !== null;
-    }
-
-    deleteUser = () => {
-        let user = this.auth.currentUser;
-        if (user) {
-            user.delete().then(res => {
-              alert('User removed');
-            }).catch(function(error) {
-              alert('Could not delete the user');
-            });
-        }
-    }
+export const isSignedIn = () => {
+  return firebase.auth().currentUser !== null;
+}
 
-    addModule = (moduleJson) => {
-        // Get a key for a new set.
-        let newModuleKey = this.db.ref().child('modules').push().key;
-    
-        let updates = {};
-        updates['/modules/' + newModuleKey] = moduleJson;
-        // User modules is used to connect the many-to-many relationship of users to modules attempted
-        updates['/user-modules/' + moduleJson.uid + '/' + newModuleKey]
-    
-        this.db.ref().update(updates);
-        return newModuleKey;
-    }
+export const deleteUser = () => {
+  let user = firebase.auth().currentUser;
+  if (user) {
+    user.delete().then(res => {
+      alert('User removed');
+    }).catch(function(error) {
+      alert('Could not delete the user');
+    });
   }
-  
-  export default Firebase;
+}
+
+export const addModule = (moduleJson) => {
+  // Get a key for a new set.
+  let newModuleKey = firebase.database().ref().child('modules').push().key;
+
+  let updates = {};
+  updates['/modules/' + newModuleKey] = moduleJson;
+  // User modules is used to connect the many-to-many relationship of users to modules attempted
+
+  // TODO this code was here but it didn't assign anything...
+  // updates['/user-modules/' + moduleJson.uid + '/' + newModuleKey]
```

Comment:
Commented this out, because it doesn't assign anything, so it was causing a compilation error. Feel free to either remove it or use this for something. I'm not sure what its purpose was so I couldn't fix it.

Reply:
Old code, good to remove